src/mito2/src/error.rs

```diff
@@ -536,6 +536,9 @@ pub enum Error {
         error: std::io::Error,
         location: Location,
     },
+
+    #[snafu(display("Failed to edit region, reason: {}", reason))]
+    EditRegion { reason: String, location: Location },
```

Comment:
Maybe you could also reuse the `InvalidRequest` error as it is a kind of request. But if you need the use another status code then it's fine to add a new error.